rsplan/primitives.py

```diff
@@ -307,9 +309,9 @@ def _interpolated(
         # straight segments) and dtheta (step size / turn radius) for curved segments.
         step = step_size if self.is_straight else step_size / self.turn_radius
 
-        seg_pts = np.arange(0, magnitude, step)
-
-        # Add segment endpoint if the list of segment points is not empty
-        seg_pts = np.append(seg_pts, [magnitude]) if seg_pts.any() else np.array([0.0])
+        # Since np.arrange does not handle floating point arithmetic well, we use
+        # linspace and the slightly awkward step of calculating the number of steps.
+        num_steps = math.ceil(magnitude / step_size)
+        seg_pts = np.linspace(0, magnitude, num_steps, endpoint=True)
 
         return seg_pts
```

Comment:
Can we just `return np.linspace(0, magnitude, num_steps, endpoint=True)`?

Reply:
yep